lib/proxysql_utils.cpp

```diff
@@ -503,3 +504,22 @@ std::pair<int,const char*> get_dollar_quote_error(const char* version) {
 		}
 	}
 }
+
+/**
+* @brief Parses a string into a long.
+* @details Parses a string into a long, with error checks. Throws an exception if parse fails.
+* @param s The string to parse.
+*
+* @return The parsed value of the string as a long.
+*/
+long parseLong(const char* s) {
+	errno = 0;
+	char *temp;
+	long val = strtol(s, &temp, 0);
+
+	if (temp == s || *temp != '\0' || ((val == LONG_MIN || val == LONG_MAX) && errno == ERANGE)) {
+		throw std::runtime_error("Could not parse long.");
```

Comment:
No need to throw an exception, error should be handled immediately, since this function is only used when parsing data received by the backend server being monitored. See the comment where this function is called.

Reply:
Removed this exception and changed the caller to return early if we detect some malformed input.